cupy/indexing/insert.py

```diff
@@ -10,4 +9,86 @@
 # TODO(okuta): Implement putmask
 
 
-# TODO(okuta): Implement fill_diagonal
+def fill_diagonal(a, val, wrap=False):
+    """Fill the main diagonal of the given array of any dimensionality.
+
+    For an array `a` with ``a.ndim > 2``, the diagonal is the list of
+    locations with indices ``a[i, i, ..., i]`` all identical. This function
+    modifies the input array in-place, it does not return a value.
+
+    Args:
+        a (cupy.ndarray): The array, at least 2-D.
+        val (scalar): The value to be written on the diagonal.
+            Its type must be compatible with that of the array a.
+        wrap (bool): If specified, the diagonal is "wrapped" after N columns.
+            This affects only tall matrices.
+
+    Examples
```

Comment:
This means that I should simplify the documentation like the other ones in cupy, right?